src/tree/SubscriptionTreeItem.ts

```diff
@@ -38,6 +38,26 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         }
     }
 
+    public async pickAppResource(context: IActionContext, options?: PickAppResourceOptions): Promise<AppResourceTreeItem> {
+        await this.getCachedChildren(context);
+        let appResources = GroupTreeItemBase.filterResources(this.cache.appResources);
+
+        if (options?.type) {
```

Comment:
I forget, do we actually change the type property on the app resource? Or do we need to check kind to distinguish between function and web apps?

Reply:
I could change the options to this, then we can use the utils we already have setup to handle this.

```ts
export interface PickAppResourceOptions extends IAzureQuickPickOptions {
    /**
     * Options to filter the picks
     */
    filter?: {
        /**
         * App resource type to filter the app resource picks
         */
        type: string;
        /**
         * App resource kind to filter the app resource picks
         */
        kind?: string;
    }

    /**
     * Set this to pick a child of the selected app resource
     */
    expectedChildContextValue?: string | RegExp | (string | RegExp)[];
}
```

Reply:
Or I was just going to pass in `microsoft.web/functionapp` for functions since it's a constant and that's what we convert function apps to anyway in resource groups using the utils in azureUtils.ts.